staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go

```diff
@@ -223,7 +223,9 @@ func convertPodNamedPortToNumber(ports []string, pod corev1.Pod) ([]string, erro
 	var converted []string
 	for _, port := range ports {
 		localPort, remotePort := splitPort(port)
-
+		if remotePort == "" {
+			return nil, fmt.Errorf("remote port cannot be empty")
+		}
 		containerPortStr := remotePort
 		_, err := strconv.Atoi(remotePort)
 		if err != nil {
```

Comment:
the problem seems to be that if the port is empty, the conversion to int fails https://go.dev/play/p/sGjQQezaGm5

Then it thinks is an empty string and tries to find the port by name, and the first container that does not have a named port matches ...

it does not seem correct